src/HonzaBotner.Discord/Command/IChatCommand.cs

```diff
@@ -7,6 +9,8 @@ namespace HonzaBotner.Discord.Command
 {
     public interface IChatCommand
     {
+        List<string> RequiredPermissions { get; }
```

Comment:
Is this list modifiable from outside? If not consider usage of ImmutableList or IReadOnly[Collection|List]. If is, then expose only the IList interface.